### PR TITLE
Final draft updates

### PR DESCRIPTION
**Pull request by Ralith**
Still TODO:
- Audit persistent congestion logic. I'm not sure we're correctly considering only ack-eliciting packets here, or waiting for an RTT sample.

### CODE REVIEW

File: quinn-proto/src/crypto/rustls.rs
Comment (Matthias247) [inline quinn-proto/src/crypto/rustls.rs:59]: Can this support both versions for a while? E.g. msquic and picoquic as a server currently support both accepting v1 and draft connections. As a client they allow to specify the version number to send. 

I think there are also more salts than just this  one.

Comment (djc) [reply quinn-proto/src/crypto/rustls.rs:59]: Maybe, but supporting multiple versions in parallel like that is probably a separate project/PR.

Comment (Matthias247) [reply quinn-proto/src/crypto/rustls.rs:59]: Maybe - but unless this is tackled this will immediately break the project for anyone who needs to interact with a draft version - which are currently e.g. all browsers.

I don't think "quinn was usable, but will stop working for the next couple of month until others have caught up" is a great story.

**Comment by Ralith** [comment]
I'm not opposed to supporting multiple versions, but that may significantly expand the scope of the PR; we've never bothered to straddle breaking changes before.

**Comment by djc** [comment]
I think the time is right for multiple version support, I'd just prefer to decouple it a bit from these updates in terms of when/how we land it. It'd be neat to do it first in a separate PR that adds most of the required mechanics with draft 29-32 support, and then we can add a 34 option in this PR.

**Comment by Ralith** [comment]
Multiple version support is blocked on https://github.com/ctz/rustls/pull/544.

**Comment by Matthias247** [comment]
> It'd be neat to do it first in a separate PR that adds most of the required mechanics with draft 29-32 support, and then we can add a 34 option in this PR.

 Correct me if I'm wrong, but I thought all drafts (incl 34) use the old salts and are already compatible. Only the non draft version (quic v1) uses the new salts and requires extra code to be compatible.

**Comment by Ralith** [comment]
Yeah, we should be able to have 29..=34 trivially.

**Comment by djc** [comment]
Even so, we can still separate the implementation of a mechanism to support different-salt versions from the inclusion of a particular version that actually has a different salt, right?

**Comment by Matthias247** [comment]
Sure! Would be nice to do this before merging the new salts, to avoid breaking draft version users.

**Comment by Ralith** [comment]
Moving non-breaking changes into https://github.com/quinn-rs/quinn/pull/1084; multi-version support can then be implemented in another separate PR, followed by the breaking salt and version number changes.

**Review by djc — APPROVED** [review]

**Pull request merged**
